src/Foundatio.Aliyun/Storage/AliyunFileStorage.cs

```diff
@@ -37,10 +38,30 @@ public class AliyunFileStorage : IFileStorage {
             return false;
         }
 
+        private Exception UnwrapException(Exception ex) {
+            if (ex is AggregateException aggregateException && aggregateException.InnerExceptions.Count == 1) {
+                return aggregateException.InnerExceptions[0];
+            }
+            return ex;
+        }
+
         private string NormalizePath(string path) {
             return path?.Replace('\\', '/');
         }
 
+        private bool DoesBucketExist(string bucketName) {
+            try {
+                return _client.DoesBucketExist(bucketName);
+            }
+            catch (Exception ex) when (IsNotFoundException(ex)) {
+                return false;
+            }
+            catch (Exception ex) {
+                ExceptionDispatchInfo.Capture(UnwrapException(ex)).Throw();

```

Comment:
Were you seeing an issue where this catch block was needed? Perhaps we should just return IsNotFoundException inside of the first try catch and `throw` if it's not. Does the ops client return aggregate exceptions?